src/mongocxx/test/collection_mocked.cpp

```diff
@@ -244,7 +244,8 @@ TEST_CASE("Collection", "[collection]") {
             REQUIRE(skip == skip);
             REQUIRE(limit == limit);
             REQUIRE(batch_size == batch_size);
-            REQUIRE(bson_get_data(query) == doc.data());
+            bsoncxx::document::view query_view{bson_get_data(query), query->len};
```

Comment:
For simplicity, we now send all finds with the `{$query: {}}` form. Previously we would simply send the match document when we didn't have modifiers to the find. This test assumed that we would never send finds with the $query syntax, and would have broken had we ever tested sorts (which this PR implements) or modifiers.

Since we _always_ use the $query form now, this new assumption is safe (and works when testing sorts). I will add a test to the mock as well to ensure this stays functioning.